jwst/resample/resample_spec.py

```diff
@@ -144,31 +208,71 @@ def build_nirspec_output_wcs(self, refmodel=None):
         # save the wcs of the reference model
         refwcs = refmodel.meta.wcs
 
-        # setup the transforms that are needed
+        # set up the transforms that are needed
         s2d = refwcs.get_transform('slit_frame', 'detector')
         d2s = refwcs.get_transform('detector', 'slit_frame')
         s2w = refwcs.get_transform('slit_frame', 'world')
+        w2s = refwcs.get_transform('world', 'slit_frame')
 
         # estimate position of the target without relying on the meta.target:
         # compute the mean spatial and wavelength coords weighted
         # by the spectral intensity
         bbox = refwcs.bounding_box
         grid = wcstools.grid_from_bounding_box(bbox)
         _, s, lam = np.array(d2s(*grid))
-        sd = s * refmodel_data
-        ld = lam * refmodel_data
-        good_s = np.isfinite(sd)
-        total = np.sum(refmodel_data[good_s])
-        if np.any(good_s) and total != 0:
-            wmean_s = np.sum(sd[good_s]) / total
-            wmean_l = np.sum(ld[good_s]) / total
+
+        # find invalid values
+        good = np.isfinite(s) & np.isfinite(lam) & np.isfinite(refmodel_data)
+        refmodel_data[~good] = np.nan
+
+        # reject the worst outliers in the data
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=AstropyUserWarning,
+                                    message=".*automatically clipped.*")
+            weights = sigma_clip(refmodel_data, masked=True, sigma=100.0)
+        weights = np.ma.filled(weights, fill_value=0.0)
+        if not np.all(weights == 0.0):
+            wmean_s = np.average(s[good], weights=weights[good])
+            wmean_l = np.average(lam[good], weights=weights[good])
```

Comment:
I noticed in testing that the target RA and Dec was sometimes pulled away from the true spectral trace by enormous outliers in the spectral image, because the target position was computed from a simple average.  These changes were to fix that issue for at least the most egregious cases.